meteor/client/lib/ShowStyleVariantImportButton.tsx

```diff
@@ -0,0 +1,168 @@
+					)
+					if (!Array.isArray(newShowStyleVariants)) {
+						throw new Error('Imported file did not contain an array')
+					}
+				} catch (error) {
+					NotificationCenter.push(
+						new Notification(
+							undefined,
+							NoticeLevel.WARNING,
+							t('Failed to import new showstyle variants: {{errorMessage}}', { errorMessage: error + '' }),
+							'VariantSettings'
+						)
+					)
+					return
+				}
+
+				this.importShowStyleVariantsFromArray(newShowStyleVariants)
+			}
+			reader.readAsText(file)
+		}
+
+		private importShowStyleVariantsFromArray(importedShowStyleVariants: ShowStyleVariant[]): void {
+			importedShowStyleVariants.forEach((showStyleVariant: ShowStyleVariant, index: number) => {
+				const rank = this.props.showStyleVariants.length
+				showStyleVariant._rank = rank + index
+				if (this.showStyleVariantAlreadyExists(showStyleVariant)) {
+					this.provideImportOptions(showStyleVariant)
+				} else {
+					this.importShowStyleVariant(showStyleVariant)
+				}
+			})
+		}
+
+		private showStyleVariantAlreadyExists(importedShowStyleVariant: ShowStyleVariant): boolean {
+			const exists = this.props.showStyleVariants.find(
+				(variant: ShowStyleVariant) => variant.name === importedShowStyleVariant.name
+			)
+			return !!exists
+		}
+
+		private provideImportOptions(showStyleVariant: ShowStyleVariant) {
+			doModalDialog({
+				title: t('Do you want to replace this variant?'),
+				yes: t('Replace'),
+				no: t('Keep both variants'),
+				onAccept: () => {
+					this.replaceShowStyleVariant(showStyleVariant)
+				},
+				onSecondary: () => {
+					const resemblingNameCount = this.getDuplicatedShowStyleVariantAmount(showStyleVariant)
+					showStyleVariant.name += ' (' + resemblingNameCount + ')'
+					this.importShowStyleVariant(showStyleVariant)
+				},
+				message: (
+					<React.Fragment>
+						<p>
+							{t('Do you want to replace "{{showStyleVariantName}}"?', {
+								showStyleVariantName: showStyleVariant.name,
+							})}
+						</p>
+					</React.Fragment>
+				),
+			})
+		}
+
+		private importShowStyleVariant(showStyleVariant: ShowStyleVariant) {
+			MeteorCall.showstyles.importShowStyleVariantAsNew(showStyleVariant).catch(() => {
+				NotificationCenter.push(
+					new Notification(
+						undefined,
+						NoticeLevel.WARNING,
+						t('Failed to import Variant {{name}}.', {
+							name: showStyleVariant.name,
+						}),
+						'VariantSettings'
+					)
+				)
+			})
+		}
+
+		private replaceShowStyleVariant(newShowStyleVariant: ShowStyleVariant) {
+			const existingShowStyleVariant = this.props.showStyleVariants.find((variant: ShowStyleVariant) => {
+				return variant.name === newShowStyleVariant.name
+			})
+			if (existingShowStyleVariant) {
+				MeteorCall.showstyles
+					.removeShowStyleVariant(existingShowStyleVariant._id)
+					.then(() => {
+						MeteorCall.showstyles.importShowStyleVariantAsNew(newShowStyleVariant).catch(logger.warn)
+					})
+					.catch(logger.warn)
```

Comment:
I think it can be written like that
```suggestion
					.then(() => (MeteorCall.showstyles.importShowStyleVariantAsNew(newShowStyleVariant)))
					.catch(logger.warn)
```